oneflux_steps/ustar_cp_python/cpd_evaluate_functions.py

```diff
@@ -325,11 +333,345 @@ def computeTemperatureThresholds(
+
+    - Shallow copies are made where needed to avoid modifying the original data.
+    - The switch-case in MATLAB is replaced by Python if/elif statements.
+
+    Example
+    -------
+    >>> t = np.linspace(737060.0, 737389.0, 24*365/2)  # Example time array
+    >>> nt, m, EndDOY, nPerBin, nN = initializeParameters(t, 4, 2, 3)
+    """
+
+    # Make a shallow copy of t
+    tCopy = t.copy()
+
+    # Derive basic time information
+    nt = len(tCopy)
+    y, m, d, h, mn, s = fcDatevec(tCopy)  
+    iYr = int(np.median(y))          # median year
+    EndDOY = fcDoy(datenum(iYr, 12, 31.5))  
+
+    # Estimate sampling frequency
+    # e.g., daily data => nPerDay=1, or 30-min data => nPerDay=48, etc.
+    nPerDay = int(round(1 / np.nanmedian(np.diff(tCopy))))
+
+    # Define default
+    nPerBin = 5
+
+    # Adjust binning based on sampling frequency
+    if nPerDay == 24:
+        nPerBin = 3
+    elif nPerDay == 48:
+        nPerBin = 5
+
+    # Number of points per season for 'n' dimension
+    nPerSeasonN = nStrataN * nBins * nPerBin
+    nN = nSeasons * nPerSeasonN
+
+    return nt, m, EndDOY, nPerBin, nN
+
+
+def initializeStatistics(nSeasons: int, nStrataX: int) -> Tuple[Any, Any]:
+    """
+    initializeStatistics
+
+    Creates two 2D arrays (Stats2, Stats3) of the same shape [nSeasons x nStrataX].
+    Each entry in these arrays is a shallow copy of a base stats object (StatsMT)
+    generated by generate_statsMT(). This replicates the MATLAB code logic, but uses
+    Python's 0-based indexing and shallow copies.
+
+    Parameters
+    ----------
+    nSeasons : int
+        Number of seasons.
+    nStrataX : int
+        Number of strata for the x dimension.
+
+    Returns
+    -------
+    Stats2 : 2D list (or array) of shape [nSeasons x nStrataX]
+        Each element is a shallow copy of StatsMT.
+    Stats3 : 2D list (or array) of shape [nSeasons x nStrataX]
+        Each element is a shallow copy of StatsMT.
+
+    Notes
+    -----
+    - Shallow copies are made to ensure each element in Stats2/Stats3 can be updated
+      independently if needed.
+    """
+
+    # Retrieve the base stats object (placeholder function).
+    StatsMT = generate_statsMT()
+
+    # Construct Stats2, Stats3 as 2D arrays (lists of lists here)
+    # Each element is a shallow copy of StatsMT
+    Stats2 = [[copy.copy(StatsMT) for _ in range(nStrataX)] for _ in range(nSeasons)]
+    Stats3 = [[copy.copy(StatsMT) for _ in range(nStrataX)] for _ in range(nSeasons)]
+
+    return Stats2, Stats3
+
+
+def cpdEvaluateUStarTh4Season20100901(
+    t: np.ndarray,
+    NEE: np.ndarray,
+    uStar: np.ndarray,
+    T: np.ndarray,
+    fNight: np.ndarray,
+    fPlot: int,
+    cSiteYr: str
+) -> Tuple[np.ndarray, Any, np.ndarray, Any]:
+    """
+    cpdEvaluateUStarTh4Season20100901
+
+    Estimates uStarTh for one year of data using change-point detection (cpd) methods
+    within the general framework of the Papale et al. (2006) uStarTh evaluation method.
+
```

Comment:
Would be good to preserve in here that this was original written by Alan Barr.
```suggestion
    Originally written (in MATLAB) by Alan Barr 15 Jan 2010.
```